config.go

```diff
@@ -1917,6 +1925,52 @@ func parseRPCParams(cConfig *lncfg.Chain, nodeConfig interface{},
 	return nil
 }
 
+// extractEnvVariable extracts the value of an environment variable from a string.
+// It supports the following formats:
+// 1) $ENVIRONMENT_VARIABLE
+// 2) ${ENVIRONMENT_VARIABLE}
+// 3) ${ENVIRONMENT_VARIABLE:-DEFAULT_VALUE}
+//
+// Standard environment variable naming conventions:
+// - ENVIRONMENT_VARIABLE contains letters, digits, and underscores, and does not start with a digit.
+// - DEFAULT_VALUE follows the rule that it can contain any characters except whitespace.
+//
+// Parameters:
+// - value: The input string containing references to environment variables.
+//
+// Returns:
+//   - string: The value of the specified environment variable, or the default value
+//     if provided, or the original input string if no matching variable is found or set.
+func extractEnvVariable(value string) string {
+	re := regexp.MustCompile(`^(?:\${([a-zA-Z_][a-zA-Z0-9_]*)?(:-(\S+))?})|(?:\$([a-zA-Z_][a-zA-Z0-9_]*))$`)
+
+	matches := re.FindStringSubmatch(value)
+
+	if len(matches) > 0 {
+		// envVariable, Group 1 corresponding to this format: ${ENVIRONMENT_VARIABLE}
+		// or ${ENVIRONMENT_VARIABLE:-DEFAULT_VALUE}
+		envVariable := matches[1]
+		// envVariable, Group 4 corresponding to this format: $ENVIRONMENT_VARIABLE
+		if matches[4] != "" {
+			envVariable = matches[4]
+		}
+
+		// defaultValue, Group 3 corrosponding to this format: ${ENVIRONMENT_VARIABLE:-DEFAULT_VALUE}
+		defaultValue := matches[3]
+
+		envValue := os.Getenv(envVariable)
+
+		// Use the default value if the environment variable is not set
+		if envValue == "" && defaultValue != "" {
```

Comment:
I think the second clause of this check is unnecessary
```suggestion
		if envValue == "" {
```